ddapm_test_agent/trace.py

```diff
@@ -47,6 +51,19 @@
 MetricType = Union[int, float]
 
 
+def convert_numbers(data):
+    if isinstance(data, list):
+        return [convert_numbers(item) for item in data]
+    elif isinstance(data, dict):
+        return {key: convert_numbers(value) for key, value in data.items() if key != "meta"}
```

Comment:
the only dict we'd expect that isn't `meta` would be `metrics`. Why do we need to convert the values in `metrics` to numeric types?